packages/rx_bloc_cli/mason_templates/bricks/feature_onboarding/__brick__/lib/feature_onboarding/ui_components/phone_number_picker.dart

```diff
@@ -0,0 +1,191 @@
+                              ),
+                            ),
+                          ),
+                        )
+                      : const SizedBox.shrink(),
+            ),
+          ],
+        ),
+      );
+
+  /// region Builders
+
+  Widget _buildPhoneNumberPrefix(
+    BuildContext context,
+    CountryCodeModel? selectedCountryCode,
+  ) =>
+      TextButton(
+        onPressed: () => _onCountryCodeSegmentPressed(
+          context,
+          selectedCountryCode,
+        ),
+        style: TextButton.styleFrom(
+          padding: const EdgeInsets.symmetric(vertical: 16),
+          shape: const RoundedRectangleBorder(
+            borderRadius: BorderRadius.only(
+              topLeft: Radius.circular(_borderRadius),
+              bottomLeft: Radius.circular(_borderRadius),
+              topRight: Radius.zero,
+              bottomRight: Radius.zero,
+            ),
+          ),
+          backgroundColor: countryCodeBgColor,
+        ),
+        child: Text(
+          '+${selectedCountryCode?.code ?? countryCodeHint}',
+          style: countryCodeTextStyle ??
+              TextStyle(
+                color: context.designSystem.colors.primaryColor,
+                fontSize: 16.0,
+              ),
+        ),
+      );
+
+  Widget _buildPhoneNumberInput(
+    BuildContext context,
+    String number,
+    String hint,
+  ) =>
+      TextField(
+        onChanged: (value) => context
+            .read<OnboardingPhoneBlocType>()
+            .events
+            .setPhoneNumber(value),
+        keyboardType: TextInputType.phone,
+        inputFormatters: [PhoneNumberFormatter()],
+        decoration: InputDecoration(
+          isDense: true,
+          hintText: hint,
+          contentPadding: const EdgeInsets.fromLTRB(16, 16, 8, 16),
+          border: const OutlineInputBorder(
+            borderRadius: BorderRadius.only(
+              topRight: Radius.circular(_borderRadius),
+              bottomRight: Radius.circular(_borderRadius),
+              topLeft: Radius.zero,
+              bottomLeft: Radius.zero,
+            ),
+          ),
+        ),
+      );
+
+  /// endregion
+
+  /// region Callbacks and helpers
+
+  void _onCountryCodeSegmentPressed(
+    BuildContext context,
+    CountryCodeModel? selectedCountryCode,
+  ) =>
+      showSearchPickerBottomSheet<CountryCodeModel>(
+        context: context,
+        title: context.l10n.featureOnboarding.selectCountry,
+        hintText: context.l10n.featureOnboarding.typeSubstring,
+        retryText: context.l10n.featureOnboarding.retry,
+        selectedItem: selectedCountryCode,
+        onItemTap: (item) {
+          if (item == null) return;
+          context.read<OnboardingPhoneBlocType>().events.setCountryCode(item);
+        },
+        service: context.read<SearchCountryCodeService>(),
+        emptyBuilder: () => Padding(
+          padding: const EdgeInsets.symmetric(horizontal: 10),
+          child: MessagePanelWidget(
```

Comment:
Please confirm the common UI component from the base folder is used

Reply:
This is a widget from the `widget_toolkit` package. However, this approach was replaced with another one.